packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/mod.rs

```diff
@@ -46,31 +46,55 @@ impl<C: AsClient> PostgresStore<C> {
         entity_vertex_id: EntityVertexId,
         dependency_context: &'a mut DependencyContext,
         subgraph: &'a mut Subgraph,
-        current_resolve_depth: GraphResolveDepths,
+        mut resolve_depths: GraphResolveDepths,
+        time_projection: TimeProjection,
     ) -> Pin<Box<dyn Future<Output = Result<(), QueryError>> + Send + 'a>> {
         async move {
-            let dependency_status = dependency_context
-                .knowledge_dependency_map
-                .insert(&entity_vertex_id, current_resolve_depth);
-
             let time_axis = subgraph.resolved_time_projection.image_time_axis();
 
-            let entity: &Entity = match dependency_status {
-                DependencyStatus::Unresolved => {
-                    match entity_vertex_id.subgraph_vertex_entry(subgraph) {
-                        RawEntryMut::Occupied(entry) => entry.into_mut(),
-                        RawEntryMut::Vacant(_) => {
-                            // Entities are always inserted into the subgraph before they are
-                            // resolved, so this should never happen. If it does, it is a bug.
-                            unreachable!("entity should already be in the subgraph")
-                        }
-                    }
+            let entity: &Entity = match entity_vertex_id.subgraph_vertex_entry(subgraph) {
+                RawEntryMut::Occupied(entry) => entry.into_mut(),
+                RawEntryMut::Vacant(_) => {
+                    // Entities are always inserted into the subgraph before they are resolved, so
+                    // this should never happen. If it does, it is a bug.
+                    unreachable!("entity should already be in the subgraph")
                 }
+            };
+
+            let version_interval = entity
+                .metadata()
+                .version()
+                .projected_time(time_axis)
+                .into_time_interval();
+
+            let Some(mut time_projection) = time_projection.intersect_image(version_interval) else {
+                // The entity is not in the time projection, so we don't need to resolve it.
+                // This should never happen as the time projection is based on this entity's
+                // version interval, but we emit a warning just in case.
+                tracing::warn!(
+                    entity_id = ?entity_vertex_id.base_id(),
+                    "entity is not in the time projection"
+                );
```

Comment:
`traverse_entity` is called with the returned entities from `read` with `time_projection`. This implies, that the version interval of `entity` is overlapping with `time_projection`. `intersect_image` returns `None` if there are no overlapping points, so this should never happen. I'm happy to make this an error.

Reply:
```suggestion
                // `traverse_entity` is called with the returned entities from `read` with
                // `time_projection`. This implies, that the version interval of `entity` is
                // overlapping with `time_projection`. `intersect_image` returns `None` if there are
                // no overlapping points, so this should never happen.
                unreachable!("the version interval of the entity does not overlap with the time projection");
```

Reply:
If we're confident this would be a bug, then the suggestion seems fine. Ideally we'd have _some_ confidence that automated tests would pick it up if that were the case